client/client_test.go

```diff
@@ -155,17 +158,18 @@ func TestRemoveProbe(t *testing.T) {
 
 // TestListProbes tests that the ListProbes() method returns all active probes in Cloudprober
 func TestListProbes(t *testing.T) {
+	ctx := context.Background()
 	client := setupCloudproberAndClient(t)
 	defer teardownCloudproberAndClient(t, client)
 
 	for i := 0; i < testRunCount; i++ {
 		probeName := fmt.Sprintf("testExtension%d", i)
-		if err := client.RegisterAndAddProbe(context.Background(), int(myprobe.E_RedisProbe.Field), generateRedisProbeDef(probeName), &myprobe.Probe{}); err != nil {
+		if err := client.RegisterAndAddProbe(ctx, int(myprobe.E_RedisProbe.Field), generateRedisProbeDef(probeName), &myprobe.Probe{}); err != nil {
 			t.Errorf("Probe not correctly registered and added to Cloudprober, expected error: %v, got: %v", nil, err)
```

Comment:
Replace Errorf with Fatalf.
Rationale: If RegisterAndAddProbe fails, it means there's a problem with the environment or the test code itself. There's no point in continuing the test.
In general test set-up/tear-down should use Fatalf. Check and fix this throughout the codebase.

Reply:
While `RegisterAndAddProbe()` is a setup for testing `RemoveProbe()` and `ListProbes()`, it is still one of the methods under test in this package, and it is very useful to not exit the test the first time that it fails. 

I would prefer to keep the `Errorf()` for calls to add, remove and list probes. It can be helpful to see if a method fails throughout the remainder of the test, just because it failed once.